### PR TITLE
Add GH Actions Build

### PR DESCRIPTION
**Pull request by AndyTWF**

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        run: |
+          cd msvc
+          nuget restore
+          cd ..
+
+      # If we're running on main, get the next release version
+      - name: Setup Build Version
+        if: github.ref == 'refs/heads/main'
+        run: |
+          $semanticRelease = npx semantic-release@15 --dry-run | Select-String -Pattern "Published release ((0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?)"
```
Comment (AxonC) [inline .github/workflows/build.yml:54]: Holy regex batman!

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+      - name: Setup Boost Environment
+        run: |
+          echo "BOOST_ROOT=${env:BOOST_ROOT_1_72_0}\\include" | Out-File -FilePath $env:GITHUB_ENV -Encoding utf8 -Append
+          echo "BOOST_LIBRARYDIR=${env:BOOST_ROOT_1_72_0}\\lib" | Out-File -FilePath $env:GITHUB_ENV -Encoding utf8 -Append
+
+      - name: Cache OpenSSL
+        id: cache-openssl
+        uses: actions/cache@v2.1.3
+        with:
+          path: "C:\\Program Files (x86)\\OpenSSL-Win32"
```
Comment (AxonC) [inline .github/workflows/build.yml:32]: Is this within the windows actions VM?

**Review by AxonC — APPROVED** [review]

File: .github/workflows/build.yml (hunk 1 above)
Comment (AndyTWF) [reply .github/workflows/build.yml:54]: Yep - semantic isn’t too great for these sorts of builds but it works :D

File: .github/workflows/build.yml (hunk 2 above)
Comment (AndyTWF) [reply .github/workflows/build.yml:32]: It’s cached at a repo level :)

**Pull request merged**